QuizApp/Quiz/QuizListView.swift

```diff
@@ -0,0 +1,96 @@
+import SwiftUI
+import Factory
+
+struct QuizListView: View {
+
+    @State private var segmentationSelection: String = "All"
+    @ObservedObject private var viewModel: QuizViewModel
+
+    init(viewModel: QuizViewModel) {
+        self.viewModel = viewModel
+        setSegmentedControlAppearance()
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading, spacing: 20) {
+                Picker("", selection: $segmentationSelection) {
+                    let items = ["All"] + CategorySection.allCases.map {$0.rawValue}
+                    ForEach(items, id: \.self) { option in
+                        Text(option)
+                    }
+                }
+                .onChange(of: segmentationSelection) { _ in
+                    getQuizzes()
+                }
+                .pickerStyle(SegmentedPickerStyle())
+                .cornerRadius(10)
+
+                let allCategories = CategorySection.allCases.map { $0.rawValue }
+                if allCategories.contains(segmentationSelection) {
+                    let section = CategorySection(rawValue: segmentationSelection)!
+                    Section(header: Text(section.rawValue)
+                        .sectionHeaderStyle(section)) {
+                            ForEach(viewModel.quizzes, id: \.self) { quiz in
+                                QuizCellView(quiz: quiz)
+                                    .onTapGesture {
+                                        viewModel.showQuizDetails(quiz: quiz)
+                                    }
+                            }
+                            .cornerRadius(30)
+                        }
+                } else {
+                    ForEach(CategorySection.allCases, id: \.self) { section in
+                        let filteredQuizzes = viewModel
+                            .quizzes
+                            .filter { $0.category == section.rawValue.uppercased() }
+                        if !filteredQuizzes.isEmpty {
+                            Section(header: Text(section.rawValue)
+                                .sectionHeaderStyle(section)) {
+                                    ForEach(filteredQuizzes, id: \.self) { quiz in
+                                        QuizCellView(quiz: quiz)
+                                            .onTapGesture {
+                                                viewModel.showQuizDetails(quiz: quiz)
+                                            }
+                                    }
+                                    .cornerRadius(30)
+                                }
+                        }
+                    }
+                }
```

Comment:
Try to move as much as possible to view model and I think you'll be able to have only one `Section` view which will be updated and filtered through ViewModel.